### DISCUSSION TITLE
How do Slack Notifications change the messages link URL in slack channel ?

### DISCUSSION THREAD

**Discussion by yatingAnne** [PrefectHQ/prefect#5545] [Q&A] [answered]
I create a prefect server use cli `prefect server start`, and create a local agent use cli` prefect agent local start`.
Then I need hooks Slack to get notified when certain state changes occur within flows or for a given task.
The Prefect hook Slack is reference [Slack Notifications](https://docs.prefect.io/core/advanced_tutorials/slack-notifications.html#installation-instructions)
This is my Prefect flow program：

```
from prefect import Flow, task
from prefect.utilities.notifications import slack_notifier
import os

os.environ["PREFECT__CONTEXT__SECRETS__SLACK_WEBHOOK_URL"] = "https://hooks.slack.com/services/XXXXX"

@task(name="1/x task", state_handlers=[slack_notifier])
def div(x):
    return 1 / x

@task(name="Add 1 task", state_handlers=[slack_notifier])
def add(x):
    return x + 1

with Flow('prefect-slack-test') as f:
    res = div(x=add(-1))

f.register(project_name="Demo")
```

Now, I can send the message  when run the flow schedule, but the message link URL has a mistake like
`http://localhost:8080/default/task-run/XXXXXX`
The URL show `http://localhost` not the `Prefect Server host (http://10.XX.XX.XX)`, and occur `Localhost Refused to Connect, ERR_CONNECTION_REFUSED`.

How can I solve this error?

**Comment by yatingAnne** [comment, marked as answer]
I find the solution that add host, port and endpoint in config.toml, like

```
[server]
    [server.ui]
    host = "http://10.XX.XX.XX"
    port = "8080"
    endpoint = "${server.ui.host}:${server.ui.port}"
```

ref: https://github.com/PrefectHQ/prefect/blob/master/src/prefect/config.toml

**Reply by mudit-alphagrep** [reply]
Not working for me still getting 

Message: Some reference tasks failed.
Link: http://localhost:4200/default/
